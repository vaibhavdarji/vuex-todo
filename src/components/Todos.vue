<template>
  <el-table
    empty-text="No Items"
    v-loading="loading"
    :show-header="false"
    :data="todos"
    style="width: 100%"
    class-name="todo-table"
    size="small">
    <el-table-column width="55">
      <template slot-scope='scope'>
        <el-checkbox v-model="scope.row.completed" size="small" :border="true" @change="onChangeTodoStatus(scope.row)"></el-checkbox>
      </template>
    </el-table-column>
    <el-table-column>
      <template slot-scope='scope'>
        <span
          v-show="scope.row.editable == false" 
          @click="handleClick(scope.row)"
          :class="[{'completed': scope.row.completed}]"
          >{{scope.row.value}}</span>
        <el-form v-show="scope.row.editable"
          :model="scope.row"
          :rules="rules"
          ref="todo"
          class="todo-item"
          @submit.native.prevent >
          <el-form-item prop="value">
            <el-input
              v-show="scope.row.editable"
              :autofocus="scope.row.editable" style="width: 80%; "
              :ref="scope.row.id.toString()"
              v-model="scope.row.value"
              @blur.prevent="hideForm('todo', scope.row)"
              @keyup.enter.native.prevent="submitForm('todo', scope.row)"
            >
            </el-input>
          </el-form-item>
        </el-form>
      </template>
    </el-table-column>
    <el-table-column width="55">
      <template slot-scope="scope">
        <i class="el-icon-close" @click="deleteTodo(scope.row)"></i>
      </template>
    </el-table-column>
  </el-table>
</template>

<script>
import { mapGetters, mapActions, mapMutations } from 'vuex';
export default {
  name: 'Todos',
  data() {
    return {
      rules: {
        value: [
          { required: true, message: 'Please type your todo', trigger: 'blur' },
          { min: 3, message: 'Length at least 3 characters', trigger: 'blur' }
        ]
      }
    };
  },
  /*components: {
    TodoItem
  },*/

  computed: {
    ...mapGetters(['todos', 'loading'])
  },
  mounted() {
    this.getTodo();
  },
  methods: {
    ...mapActions({
      getTodo: 'GET_TODO',
      removeTodo: 'REMOVE_TODO',
      updateTodo: 'UPDATE_TODO'
    }),

    onChangeTodoStatus(todo) {
      this.updateTodo(todo);
    },

    handleClick(todo) {
      todo.editable = !todo.editable;
      this.$nextTick(() => {
        this.$refs[todo.id.toString()].$el.querySelector('input').focus();
      });
    },

    submitForm(formName, todo) {
      this.$refs[formName].validate(valid => {
        if (!valid) {
          return false;
        }
        todo.editable = false;
        this.updateTodo(todo);
      });
    },

    hideForm(formName, todo) {
      this.$refs[formName].validate(valid => {
        if (!valid) {
          return false;
        }
        todo.editable = false;
      });
    },

    deleteTodo(todo) {
      this.$confirm(
        'This will permanently delete the file. Continue?',
        'Warning',
        {
          confirmButtonText: 'OK',
          cancelButtonText: 'Cancel',
          type: 'warning'
        }
      )
        .then(() => {
          this.removeTodo(todo);
        })
        .catch(() => console.log('canceled'));
    }
  }
};
</script>


