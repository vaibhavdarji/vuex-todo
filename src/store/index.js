import Vue from 'vue';
import Vuex from 'vuex';
import * as actions from '../actions';
import * as mutations from '../mutations';
Vue.use(Vuex);

const state = {
  loading: true,
  todos: []
};

const getters = {
  todos: state => state.todos,
  loading: state => state.loading
};

const store = new Vuex.Store({
  state,
  getters,
  actions,
  mutations
});

export default store;
