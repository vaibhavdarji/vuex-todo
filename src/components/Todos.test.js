import { shallow, shallowMount } from 'vue-test-utils';
import Todos from './Todos';
import store from '../store';
// import { GET_TODO } from '../actions';

describe('Todos', () => {
  it('renders', () => {
    store.replaceState({ todos: [{}] });
    store.dispatch = jest.fn();
    const wrapper = shallow(Todos, { store });
    expect(wrapper.element).toMatchSnapshot();
  });
});
