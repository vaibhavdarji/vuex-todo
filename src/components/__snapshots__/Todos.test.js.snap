// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todos renders 1`] = `
<el-table
  class-name="todo-table"
  data="[object Object]"
  empty-text="No Items"
  size="small"
  style="width: 100%;"
>
  <el-table-column
    width="55"
  />
   
  <el-table-column />
   
  <el-table-column
    width="55"
  />
</el-table>
`;
